rust/src/io/object_store.rs

```diff
@@ -167,4 +167,9 @@ impl ObjectStore {
             Err(e) => Err(e.into()),
         }
     }
+
+    /// Get file size.
+    pub async fn size(&self, path: &Path) -> Result<usize> {
+        Ok(self.inner.head(path).await?.size)
```

Comment:
might be slow if it's s3 path and this gets called a bunch of times? ok for now as it's a utility for testing. add a todo here or something?

Reply:
i am not sure what optimization we can do here. And getting metadata on s3 is slow is well-understood by users, they should have the expectation?

Unlikely the upstream object store crate will explicitly call out that operations on S3 is slow but on local filesystem is faster?